scripts/check_integrity.py

```diff
@@ -0,0 +1,251 @@
+def check_string_fields(entry):
+    for field in string_fields:
+        if is_set('string_' + field):
+            if field in entry and isinstance(entry[field], str):
+                add_problem(entry, 'non_string_' + field, entry[field])
+
+
+## Creates bibtex strings for fields that are declared as string fields but do not contains string values
+def gen_string_fields(entry):
+    for field in string_fields:
+        if field in entry and isinstance(entry[field], str):
+            field_value = entry[field]
+            key = re.sub('[^A-Z]', '', field_value)
+            new_strings[field_value] = (field, key, field_value)
+
+
+## Checks that required, wanted or optional fields exists for a given entry or is declared as missing
+def check_field_type_exists(entry, field_type):
+    if is_set(field_type + '_fields'):
+        entry_type=entry['ENTRYTYPE']
+        if entry_type in fields:
+            required_fields = (fields[entry_type])[field_type] or []
+        else:
+            required_fields = []
+        entry_id = entry['ID']
+        entry_fields = set(entry.keys())
+        if 'missing' in entry:
+            missing = [x for x in entry['missing'].get_value().split(',') if x]
+        else:
+            missing = []
+        for field in required_fields:
+            if isinstance(field, list):
+                present = False
+                for alternative_field in field:
+                    if alternative_field in entry_fields or alternative_field in missing:
+                        present = True
+                        break
+                if not present:
+                    add_problem(entry, 'missing_' + field_type + '_field', field)
+            else:
+                if field not in missing and field not in entry_fields:
+                    add_problem(entry, 'missing_' + field_type + '_field', field)
+
+
+def check_field_types_exists(entry):
+    check_field_type_exists(entry, 'required')
+    check_field_type_exists(entry, 'wanted')
+    check_field_type_exists(entry, 'optional')
+
+
+## Checks that the title of an entry is protected and correctly capitalized (may produce false optional results)
+def check_title(entry):
+    if ('title' in entry):
+        org = entry['title']
+        if is_set('title_protect'):
+            if (not title_regex.match(org)):
+                add_problem(entry, 'incorrect_title_protect', org)
+        if is_set('title_capitilization'):
+            correct_title = titlecase(org)
+            if (org != correct_title):
+                add_problem(entry, 'incorrect_title_capitilization', org + " -> " + correct_title)
+
+
+## Checks that the pages of an entry are formated correctly
+def check_pages(entry):
+    if ('pages' in entry):
+        org = entry['pages']
+        if is_set('pages'):
+            if (not pages_regex.match(org)):
+                add_problem(entry, 'incorrect_pages', org)
+
+
+## Checks that the authors of an entry are in the correct ordering
```

Comment:
How can this script determine the correct order? ;) I guess it cross-checks the author order in the `authors` entry with the key?

Reply:
Yes, this comment is confusing. It checks that the author names follow the syntax `last name, first name`.
Will be changed.